sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/PersistentTokenCacheImpl.java

```diff
@@ -0,0 +1,144 @@
+                    switch (msalClientType) {
+                        case CONFIDENTIAL:
+                            persistenceSettings = getConfidentialClientPersistenceSettings();
+                            break;
+                        case PUBLIC:
+                            persistenceSettings = getPublicClientPersistenceSettings();
+                            break;
+                        default:
+                            persistenceSettings = getPublicClientPersistenceSettings();
+                            break;
+                    }
+                    cacheAccessAspect = new PersistenceTokenCacheAccessAspect(persistenceSettings);
+                    return Mono.just(true);
+                } catch (Throwable t) {
+                    return Mono.error(logger.logExceptionAsError(new ClientAuthenticationException(
+                        "Shared token cache is unavailable in this environment.", null, t)));
+                }
+            });
+    }
+
+    public void beforeCacheAccess(ITokenCacheAccessContext iTokenCacheAccessContext) {
+        cacheAccessAspect.beforeCacheAccess(iTokenCacheAccessContext);
+    }
+
+    public void afterCacheAccess(ITokenCacheAccessContext iTokenCacheAccessContext) {
+        cacheAccessAspect.afterCacheAccess(iTokenCacheAccessContext);
+    }
+
+    public enum MsalClientType {
+        PUBLIC,
+        CONFIDENTIAL;
+    }
+
+    public void setMsalClientType(MsalClientType msalClientType) {
+        this.msalClientType = msalClientType;
+    }
+
+    private PersistenceSettings getPublicClientPersistenceSettings() {
+        PersistenceSettings.Builder persistenceSettingsBuilder = PersistenceSettings.builder(
+            name != null ? name : DEFAULT_PUBLIC_CACHE_FILE_NAME, DEFAULT_CACHE_FILE_PATH);
+        if (Platform.isMac()) {
+            persistenceSettingsBuilder.setMacKeychain(
+                DEFAULT_KEYCHAIN_SERVICE, name != null ? name : DEFAULT_PUBLIC_KEYCHAIN_ACCOUNT);
+            return persistenceSettingsBuilder.build();
+        } else if (Platform.isLinux()) {
+            try {
+                persistenceSettingsBuilder
+                    .setLinuxKeyring(DEFAULT_KEYRING_NAME, DEFAULT_KEYRING_SCHEMA,
+                        name != null ? name : DEFAULT_PUBLIC_KEYRING_ITEM_NAME, DEFAULT_KEYRING_ATTR_NAME,
+                        DEFAULT_KEYRING_ATTR_VALUE, null, null);
+                return persistenceSettingsBuilder.build();
+            } catch (KeyRingAccessException e) {
+                if (!allowUnencryptedStorage) {
+                    throw logger.logExceptionAsError(e);
+                }
+                persistenceSettingsBuilder.setLinuxUseUnprotectedFileAsCacheStorage(true);
+                return persistenceSettingsBuilder.build();
+            }
+        }
+        return persistenceSettingsBuilder.build();
+    }
+
+    private PersistenceSettings getConfidentialClientPersistenceSettings() {
```

Comment:
Since we have added the `name` option to `TokenCachePersisitenceOptions` we can probably get rid of the code to create distinguished names for confidential clients vs. public clients, and leave that as an exorcise for the user if they want to cache both.

Reply:
Removed it and merged it to the common persistence settings helper method.